### PR TITLE
API for blktrace tool

### PR DESCRIPTION
**Pull request by Deixx**
+ utils

Signed-off-by: Daniel Madej <daniel.madej@intel.com>

### CODE REVIEW

File: test_tools/blktrace.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    def start_monitoring(self, buffer_size: Size = None, number_of_subbuffers: int = None):
+        if self.blktrace_pid != -1:
+            raise Exception(f"blktrace already running with PID: {self.blktrace_pid}")
+
+        self.__outputDirectoryPath = Directory.create_temp_directory().full_path
+
+        drop_caches(DropCachesMode.ALL)
+
+        number_of_subbuffers = ("" if number_of_subbuffers is None
+                                else f" --num-sub-buffers = {number_of_subbuffers}")
```
Comment (Ostrokrzew) [inline test_tools/blktrace.py]: In other command there are no spaces between param and value.

File: test_utils/os_utils.py [hunk 2]
```diff
@@ -70,6 +71,13 @@ def is_kernel_module_loaded(module_name):
     return output.exit_code == 0
 
 
+def is_mounted(path: str):
```
Comment (Ostrokrzew) [inline test_utils/os_utils.py]: In test framework in 'Device' class there's a method named like this. This shouldn't cause any conflicts, but You should check it.

File: test_tools/blktrace.py (hunk 1 above)
Comment (Deixx) [reply test_tools/blktrace.py]: Done

File: test_tools/blktrace.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    F = 1 << 3  # Flush
+    S = 1 << 4  # Synchronous
+    N = 1 << 5  # None of the above
+
+
+class BlkTrace:
+    def __init__(self, device: Device, masks: BlkTraceMask = None):
+        self._mount_debugfs()
+        if device is None:
+            raise Exception("Wrong device provided (None)")
```
[the remark this answers is not in the document]
Comment (Deixx) [reply test_tools/blktrace.py]: :D

File: test_tools/blktrace.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        self.device = device
+        self.masks = "" if masks is None else f' -a {" -a ".join([m.name for m in masks])}'
+        self.blktrace_pid = -1
+        self.__outputDirectoryPath = None
+
+    @staticmethod
```
[the remark this answers is not in the document]
Comment (Deixx) [reply test_tools/blktrace.py]: It's not using anything BlkTrace class related, but it's within the class as it's not really useful elsewhere

File: test_utils/os_utils.py (hunk 2 above)
Comment (Deixx) [reply test_utils/os_utils.py]: Device.is_mounted() checks if given instance of Device is mounted - I needed more general function that would check if given path (in this case a mountpoint) is used in mounting

**Pull request merged**